README.md

```diff
@@ -99,38 +98,52 @@ See [Authorization status](https://developer.apple.com/documentation/healthkit/h
 After authorization, you can try to start reading data.
 
 ```dart
-void read(bool isRequested) async {
-    if (isRequested) {
-      final preferredUnits =
-      await HealthKitReporter.preferredUnits([QuantityType.stepCount]);
-      preferredUnits.forEach((preferredUnit) async {
-        print('preferredUnit: ${preferredUnit.identifier}');
-        final type = QuantityTypeFactory.from(preferredUnit.identifier);
-        final quantities = await HealthKitReporter.quantityQuery(
-            type, preferredUnit, _predicate);
-        print('quantity: ${quantities.map((e) => e.map).toList()}');
-        final statistics = await HealthKitReporter.statisticsQuery(
-            type, preferredUnit, _predicate);
-        print('statistics: ${statistics.map}');
-      });
-      final characteristics = await HealthKitReporter.characteristicsQuery();
-      print('characteristics: ${characteristics.map}');
-      final categories = await HealthKitReporter.categoryQuery(
-          CategoryType.sleepAnalysis, _predicate);
-      print('categories: ${categories.map((e) => e.map).toList()}');
-      final samples = await HealthKitReporter.sampleQuery(
-          QuantityType.stepCount.identifier, _predicate);
-      print('samples: ${samples.map((e) => e.map).toList()}');
-      final sources = await HealthKitReporter.sourceQuery(
-          QuantityType.stepCount.identifier, _predicate);
-      print('sources: ${sources.map((e) => e.map).toList()}');
-      final correlations = await HealthKitReporter.correlationQuery(
-          CorrelationType.bloodPressure.identifier, _predicate);
-      print('correlations: ${correlations.map((e) => e.map).toList()}');
-    } else {
-      print('error isRequested: $isRequested');
+Future<void> read(bool isRequested) async {
+  if (isRequested) {
+    final preferredUnits =
+        await HealthKitReporter.preferredUnits([QuantityType.stepCount]);
+    for (final preferredUnit in preferredUnits) {
+      print('preferredUnit: ${preferredUnit.identifier}');
+      final type = QuantityTypeFactory.from(preferredUnit.identifier);
+      final quantities = await HealthKitReporter.quantityQuery(
+        type,
+        preferredUnit.identifier,
+        _predicate,
+      );
+      print('quantity: ${quantities.map((e) => e.map).toList()}');
+      final statistics = await HealthKitReporter.statisticsQuery(
+        type,
+        preferredUnit.identifier,
```

Comment:
Also, why do API functions accept strings instead of the `QuantityType` enum? It would remove the need of calling `.identifier` everywhere and would make the API more safe since it will be called only with correct variants. @kvs-coder thoughts?

Reply:
Hi @shilangyu 

For some methods it was done on purpose in cases, where different health types could be combined. I will consider an improvement for the statisticsQuery, as in general just a single PrefferedUnit might be enough (it contains the identififer of the type)

Reply:
> where different health types could be combined

Could you elaborate? Where can this be used?